### ISSUE TITLE
Booster predict a single case is very slow, is this a bug?

### ISSUE THREAD

**Issue by fhqplzj** [Microsoft/LightGBM#1013]
I train a Booster using the following python code:
`    train, test = lgb.Dataset(X_train, y_train), lgb.Dataset(X_test, y_test)
    param = {
        'num_leaves': 63,
        'num_iterations': 800,
        'objective': 'binary',
        'learning_rate': 0.23,
        'max_bin': 63,
        'min_data_in_leaf': 20,
        'boosting': 'gbdt',
        'metric': ['auc', 'binary_error', 'binary_logloss']
    }

    bst = lgb.train(param, train, valid_sets=[test])`
after saved the Booster model and load it to do some prediction, it costs about 2.67 seconds to predict a single sample, but 51 seconds to predict 10000 samples. 
I debug the code and find that the LGBM_BoosterPredictForCSR function in c_api.cpp costs most of the time. I guess it does some initialization work in the c++ code. But I am not sure. We need the model to do some online computing, the cases come one by one. But the time cost is unacceptable, is this a bug or is it supposed to be working like this?

**Comment by StrikerRUS** [comment]
Please see #906, #856.
